src/components/GroceryItem/GroceryItem.js

```diff
@@ -47,7 +63,7 @@ const GroceryItem = ({ item }) => {
           onChange={() => (isChecked ? resetDate() : setNewDate())}
```

Comment:
If we are going to allow users to unclick an item that they did not mean to mark as purchased, then we will have to also update our reset method to reset fields back to what they previously were. I think right now, we have the reset method restting the purhcase date to null, which would mess up our calculations.